plugins/woocommerce-admin/client/products/fills/shipping-section/shipping-section-fills.tsx

```diff
@@ -72,107 +81,161 @@ const ShippingSection = () => {
 
 	return (
 		<>
-			<WooProductSectionItem
+			<WooProductSectionItem< ShippingSectionFillPropsType >
 				id={ SHIPPING_SECTION_BASIC_ID }
-				location={ TAB_SHIPPING_ID }
+				tabs={ [
+					{
+						name: TAB_SHIPPING_ID,
+						order: 1,
+						fillProps: {
```

Comment:
This shipping section is the only one fully implemented. We're using the `fillProps` to handle the fact that we want to target two different locations (variant and non-variant), but with slots with different IDs (so that they can be targeted separately).

Reply:
Honestly this feels like taking this one step to far, this to me feels like sacrificing reusability for simplicity. I had to look at this several times to figure out what exactly was going on.
I don't think we should encourage this type of usage and users should just add two different slot fill wrappers, one for the general shipping tab and one for the variation tab.
You can always move the markup inside the `ProductSectionLayout` to a re-usable component that just takes the `basicSlot` and `dimensionsSlot` as props (or children), which will keep this much more readable.

Also I think we should encourage passing `tabs`/`templates`/`sections` data as constants that can be defined outside of the component render method.
This may not work as well for `fillProps`, so it may be worth removing `fillProps` here and making `fillProps` a callback function that provides the tabName as the first param.

Reply:
Thanks @louwie17 , and I would agree overall. It was about this time I abandoned this implementation and pursued the [other approach that it looks we'll merge](https://github.com/woocommerce/woocommerce/pull/36646).